core/src/wallet/stellar/StellarLikeWallet.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ *
+ * StellarLikeWallet.cpp
+ * ledger-core
+ *
+ * Created by Pierre Pollastri on 13/02/2019.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include "StellarLikeWallet.hpp"
+#include <async/Promise.hpp>
+#include "StellarLikeAccount.hpp"
+#include "explorers/HorizonBlockchainExplorer.hpp"
+#include "synchronizers/StellarLikeBlockchainExplorerAccountSynchronizer.hpp"
+#include <wallet/common/database/AccountDatabaseHelper.h>
+#include "database/StellarLikeAccountDatabaseHelper.hpp"
+#include <api/BoolCallback.hpp>
+
+namespace ledger {
+    namespace core {
+
+        const api::WalletType StellarLikeWallet::type = api::WalletType::STELLAR;
+
+        StellarLikeWallet::StellarLikeWallet(const std::string &walletName, const api::Currency &currency,
+                                             const std::shared_ptr<WalletPool> &pool,
+                                             const std::shared_ptr<DynamicObject> &configuration,
+                                             const DerivationScheme &derivationScheme,
+                                             const StellarLikeWalletParams &params) :
+                                             AbstractWallet(walletName, currency, pool, configuration, derivationScheme),
+                                             _params(params) {
+
+        }
+
+
+        bool StellarLikeWallet::isSynchronizing() {
+            throw make_exception(api::ErrorCode::IMPLEMENTATION_IS_MISSING, "Not implemented");
+        }
+
+        std::shared_ptr<api::EventBus> StellarLikeWallet::synchronize() {
+            throw make_exception(api::ErrorCode::IMPLEMENTATION_IS_MISSING, "Not implemented");
```

Comment:
Is it normal synchronization is not implemented yet?

Reply:
It' s part of the wallet interface but not implemented anywhere. I will return nullptr to match Bitcoin implementation. But that's something to remove from the interface since it's not used.

Reply:
Fixed by 9fc1f93d568c121fe15dcca1a67a334ad0b0e035